flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -133,7 +136,23 @@ public static void deleteJobGraphInKubernetesHA(
         }
     }
 
-    public static boolean isHaMetadataAvailable(
+    public static boolean isZookeeperHaMetadataAvailable(
+            Configuration conf, CuratorFrameworkWithUnhandledErrorListener curator) {
+        try (curator) {
+            return curator.asCuratorFramework().checkExists().forPath("/") != null;
```

Comment:
The ZK client is instantiated with `HighAvailabilityOptions.HA_ZOOKEEPER_ROOT / HighAvailabilityOptions.HA_CLUSTER_ID` as the namespace for all operations (set [here](https://github.com/apache/flink/blob/c57b84921447bb0ade5e1ff77a05ebd8bbbe71b7/flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java#L267)).
This code basically tests whether the root ZNode exists.

Note that `checkExists` will return `null` if the path doesn't exist (http://www.mtitek.com/tutorials/zookeeper/cf_check_exists.php)